test/teos/e2e/conftest.py

```diff
@@ -35,6 +35,9 @@ def teosd():
     teosd_process.join()
     shutil.rmtree(".teos")
 
+    # FIXME: wait some time, otherwise it might fail when multiple e2e tests are ran in the same session. Not sure why.
+    sleep(1)
```

Comment:
Can you share the error you get? I've tried to reproduce but I cannot.

Not that it's a big deal, but if we can find the source of the issue it may be nicer.

Reply:
I get failures like these if I run all the tests with `pytest test/teos`:

```
=========================================================================================
FAILED test/teos/unit/test_api.py::test_register - AssertionError: assert '029235274c47...9e142fff932f9' == '0365239fcb5a...ea98163f74f6c'
FAILED test/teos/unit/test_api.py::test_register_top_up - AssertionError: assert '029235274c47...9e142fff932f9' == '0365239fcb5a...ea98163f74f6c'
FAILED test/teos/unit/test_api.py::test_add_appointment - AssertionError: assert 99 == 0
FAILED test/teos/unit/test_api.py::test_add_appointment_registered_no_free_slots - assert 200 == 400
```

I'd guess some unit tests are running before the e2e tests had a chance to complete cleanup (in fact, `test_api` is the first unit test module in lexicographical order).

Reply:
Ohh, gotcha. I was only trying it out running the e2e suite, not the whole test suite. Let me see if I can find what's wrong.

Reply:
Cannot reproduce, but not like it is a big deal anyway, so I guess we can leave it like this for now.